src/clj/react_tutorial_om/events.clj

```diff
@@ -21,12 +21,35 @@
        (catch Object _
          (println (:throwable &throw-context) "unexpected error"))))))
 
+
+(defn post-league-bound-players-to-slack [{:keys [best1 best2 worst1 worst2 league]}
+                                          slack-url]
+  (when-not (str/blank? slack-url)
+    (async/thread
+      (try+
+        (client/post slack-url
+                     {:form-params {:text (format "League %s:\n
+                                                  \t\tTop players:\n
+                                                  \t\t\t\t1.- %s\n
+                                                  \t\t\t\t2.- %s\n
+                                                  \t\tBottom players:\n
+                                                  \t\t\t\t1.- %s\n
+                                                  \t\t\t\t2.- %s"
+                                                  (name league) best1 best2 worst1 worst2)}
+                      :content-type :json})
+        (catch [:status 403] {:keys [request-time headers body]}
+          (println "Slack 403 " request-time headers))
+        (catch Object _
+          (println (:throwable &throw-context) "unexpected error")))))
+
+
 (defn setup-slack-loop [channel prefix slack-url]
   (go-loop []
     (when-let [[topic msg] (<! channel)]
       (println prefix ": " topic ": " msg)
       (case topic
         :league-match (post-league-result-to-slack msg slack-url)
+        :bound-players (post-league-bound-players-to-slack msg slack-url)
```

Comment:
I think if we pass a snapshot of the "db" atom (or just the league part) then an event handler will have all the info it needs

e.g.

```clj
(defn handle-league-result
  "Given an db atom, a league name and a result update the schedule and
  matches for the league and write out to file. Return the resulting
  state. Also post to slack if possible."
  [db league {:keys [winner loser winner-score loser-score] :as result} event-ch]
  (let [db-snapshot (swap! db (partial update-league-result result league))]
    (go (>! event-ch [:league-match (assoc result
                                           :league league
                                           :league-snapshot (get-in db-snapshot [:leagues league]))])))
  {:message "ok"})
```